analyzer/emerald/emerald.go

```diff
@@ -320,3 +336,174 @@ func (m *Main) queueBlockAndTransactionInserts(ctx context.Context, batch *stora
 
 	return nil
 }
+
+// prepareEventData adds non-tx event data queries to the batch.
+func (m *Main) prepareEventData(ctx context.Context, round uint64, batch *storage.QueryBatch) error {
+	events, err := m.cfg.Source.GetEventsRaw(ctx, round)
+	if err != nil {
+		return err
+	}
+
+	nonTxEvents := []*EventData{}
+	for _, e := range events.Logs {
+		// TODO: uncomment this when oasis-sdk is bumped
+		// if e.TxHash != nil {
+		// 	continue
+		// }
+		if err = common.VisitSdkEvent(e, &common.SdkEventHandler{
+			Core: func(event *core.Event) error {
+				eventData := EventData{
+					Type: apiTypes.CoreGasUsed,
```

Comment:
Well that was ... surprising :grimacing: 

Documentation and code spelunking eventually led me to https://github.com/deepmap/oapi-codegen/blob/f4cf8f9a570380c24c6ba03ae04b9393cf120692/pkg/codegen/configuration.go#L66, and to the solution I pushed as a tiny commit to this PR. It introduces compile errors because the enum values are now renamed, I'll leave that one to you :). I checked on `main` that no previous enums are affected.

Reply:
O wow that's buried deep; thanks for digging into that. Updated!